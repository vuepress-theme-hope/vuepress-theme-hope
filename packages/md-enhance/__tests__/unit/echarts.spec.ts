import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import { echarts } from "../../src/node/markdown-it/index.js";

describe("echarts", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(echarts);

  it("Should resolve echarts container with json block", () => {
    const result = markdownIt.render(
      `
::: echarts A line chart

\`\`\`json
{
  'xAxis': {
    'type': 'category',
    'data': ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
  },
  'yAxis': {
    'type': 'value'
  },
  'series': [
    {
      'data': [150, 230, 224, 218, 135, 147, 260],
      'type': 'line'
    }
  ]
}
\`\`\`

:::
`,
      {}
    );

    expect(result).toMatch(/<ECharts.*><\/ECharts>/);
    expect(result).toContain(`title="${encodeURIComponent("A line chart")}"`);
    expect(result).not.toContain('type=""');
    expect(result).toMatchSnapshot();
  });

  it("Should resolve echarts container with js block", () => {
    const result = markdownIt.render(
      `
::: echarts A line chart

\`\`\`js
module.exports = {
  xAxis: {
    type: "category",
    data: ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"],
  },
  yAxis: {
    type: "value",
  },
  series: [
    {
      data: [150, 230, 224, 218, 135, 147, 260],
      type: "line",
    },
  ],
};
\`\`\`

:::
`,
      {}
    );

    expect(result).toMatch(/<ECharts.*><\/ECharts>/);
    expect(result).toContain(`title="${encodeURIComponent("A line chart")}"`);
    expect(result).toContain('type="js"');
    expect(result).toMatchSnapshot();
  });

  it("Should resolve echarts container with javascript block", () => {
    const result = markdownIt.render(
      `
::: echarts A line chart

\`\`\`javascript
module.exports = {
  xAxis: {
    type: "category",
    data: ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"],
  },
  yAxis: {
    type: "value",
  },
  series: [
    {
      data: [150, 230, 224, 218, 135, 147, 260],
      type: "line",
    },
  ],
};
\`\`\`

:::
`,
      {}
    );

    expect(result).toMatch(/<ECharts.*><\/ECharts>/);
    expect(result).toContain(`title="${encodeURIComponent("A line chart")}"`);
    expect(result).toContain('type="js"');
    expect(result).toMatchSnapshot();
  });

  it("Should resolve echarts container with empty title and body", () => {
    const result = markdownIt.render(
      `
::: echarts

:::
`,
      {}
    );

    expect(result).toMatch(/<ECharts.*><\/ECharts>/);
    expect(result).not.toContain('title="');
    expect(result).not.toContain('type=""');
    expect(result).toMatchSnapshot();
  });

  it("Should resolve echarts fence", () => {
    const result = markdownIt.render(
      `
\`\`\`echarts:A line chart
{
  'xAxis': {
    'type': 'category',
    'data': ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
  },
  'yAxis': {
    'type': 'value'
  },
  'series': [
    {
      'data': [150, 230, 224, 218, 135, 147, 260],
      'type': 'line'
    }
  ]
}
\`\`\`
`,
      {}
    );

    expect(result).toMatch(/<ECharts.*><\/ECharts>/);
    expect(result).toContain(`title="${encodeURIComponent("A line chart")}"`);
    expect(result).not.toContain('type=""');
    expect(result).toMatchSnapshot();
  });

  it("Should resolve echarts fence with empty title and body", () => {
    const result = markdownIt.render(
      `
\`\`\`echarts
\`\`\`
`,
      {}
    );

    expect(result).toMatch(/<ECharts.*><\/ECharts>/);
    expect(result).not.toContain('title="');
    expect(result).not.toContain('type=""');
    expect(result).toMatchSnapshot();
  });
});
