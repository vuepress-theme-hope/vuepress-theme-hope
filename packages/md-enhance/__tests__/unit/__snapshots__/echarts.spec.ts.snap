// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`echarts > Should resolve echarts container with empty title and body 1`] = `"<ECharts id=\\"echarts-0\\" config=\\"eJwBAgD9/3t9AXUA+Q==\\"></ECharts>"`;

exports[`echarts > Should resolve echarts container with javascript block 1`] = `"<ECharts id=\\"echarts-0\\" config=\\"eJxtTT0LwjAQ3fsrjsxB+qlScXBxc1JwKA7BHhqoTUkTaZD+d3OpikOX9+7dvXvvoWrb4AKHTmnTwxZeEcCwG2RfhhHAuA5LYFdh8Ka0Yzxsa2FECRU7qJZxYCeLRGesg7pbor2WREdhAtmWXeh5JHAzFU/RUMzX0qOW6D1V8EzOX3FSxBzSjCDNPSRrDklWeMhXXi7jUPUX3sh2yv6k+/u4id6Mej/K\\" title=\\"A%20line%20chart\\" type=\\"js\\"></ECharts>"`;

exports[`echarts > Should resolve echarts container with js block 1`] = `"<ECharts id=\\"echarts-0\\" config=\\"eJxtTT0LwjAQ3fsrjsxB+qlScXBxc1JwKA7BHhqoTUkTaZD+d3OpikOX9+7dvXvvoWrb4AKHTmnTwxZeEcCwG2RfhhHAuA5LYFdh8Ka0Yzxsa2FECRU7qJZxYCeLRGesg7pbor2WREdhAtmWXeh5JHAzFU/RUMzX0qOW6D1V8EzOX3FSxBzSjCDNPSRrDklWeMhXXi7jUPUX3sh2yv6k+/u4id6Mej/K\\" title=\\"A%20line%20chart\\" type=\\"js\\"></ECharts>"`;

exports[`echarts > Should resolve echarts container with json block 1`] = `"<ECharts id=\\"echarts-0\\" config=\\"eJyr5lJQUK9wrMgsVrdSAHGA3JLKglQgTz05sSQ1Pb+oUl0HIp6SWJIIFI9W983PU9dRUA8pTQVR4akpYF5GKYhyK8oEUcGJJWCqNE89Fqi7FmSEeiV2e8oSc4AmwVUVpxZlpoKURYOVQRQjWW9oaqCjYGQMIoxMgIShhY6CobEpkDAxB3LNDGIh7kWyISczD2wB0AogGctVywUAXzg7QQ==\\" title=\\"A%20line%20chart\\"></ECharts>"`;

exports[`echarts > Should resolve echarts fence 1`] = `"<ECharts id=\\"echarts-0\\" config=\\"eJyr5lJQUK9wrMgsVrdSAHGA3JLKglQgTz05sSQ1Pb+oUl0HIp6SWJIIFI9W983PU9dRUA8pTQVR4akpYF5GKYhyK8oEUcGJJWCqNE89Fqi7FmSEeiV2e8oSc4AmwVUVpxZlpoKURYOVQRQjWW9oaqCjYGQMIoxMgIShhY6CobEpkDAxB3LNDGIh7kWyISczD2wB0AogGctVywUAXzg7QQ==\\" title=\\"A%20line%20chart\\"></ECharts>"`;

exports[`echarts > Should resolve echarts fence with empty title and body 1`] = `"<ECharts id=\\"echarts-0\\" config=\\"eJwBAAD//wAAAAE=\\"></ECharts>"`;
